packages/shared/src/components/tabs/TabList.tsx

```diff
@@ -1,24 +1,49 @@
 import classNames from 'classnames';
-import React, { ReactElement, useState } from 'react';
+import React, { ReactElement, useEffect, useRef, useState } from 'react';
 
 interface ClassName {
   indicator?: string;
+  item?: string;
 }
 
 export interface TabListProps {
   items: string[];
   active: string;
   onClick?: (label: string) => unknown;
   className?: ClassName;
+  autoScrollActive?: boolean;
 }
 
 function TabList({
   items,
   active,
   onClick,
   className = {},
+  autoScrollActive,
 }: TabListProps): ReactElement {
-  const [offset, setOffset] = useState<number>(undefined);
+  const [offset, setOffset] = useState<number>(0);
+  const [indicatorOffset, setIndicatorOffset] = useState<number>(undefined);
+  const currentActiveTab = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    if (autoScrollActive && currentActiveTab.current) {
+      const scrollableParent =
+        currentActiveTab.current.parentElement.parentElement;
+      const activeTabRect = currentActiveTab.current.getBoundingClientRect();
+      const scrollableParentRect = scrollableParent.getBoundingClientRect();
+
+      // only try to scroll if the active tab is not in view
+      if (
+        activeTabRect.left < scrollableParentRect.left ||
+        activeTabRect.right > scrollableParentRect.right
+      ) {
+        currentActiveTab.current.parentElement.parentElement.scrollTo({
+          left: offset,
+          behavior: 'smooth',
```

Comment:
Not 100% sure but I think `scrollTo` or some of its options do not work on safari iOS or something so just check compatibility 🙏

Reply:
found it hard to test because it keeps redirecting me to the main app.daily.dev website, but we'll merge and double check if we need to do smth differently for safari (or if there is anything we can do)